lib/src/kbasesearchengine/common/GUID.java

```diff
@@ -128,21 +152,54 @@ public GUID(String textGUID) {
             this.subObjectId = subObjectPart.substring(slashPos + 1);
         }
     }
-    
-    //TODO NOW if the version or accessGroupID is missing, how do you know which is which? Need to know storage code -> has access group mapping or have a placeholder (-1) for missing access group
+
+    /**
+     * Returns the URL-encoded version of this GUID.
+     *
+     * For the sake of readability the URL-encoded format of the GUID is,
+     *
+     *   {storageCode}___[{AccessGroupID}_]{accessGroupObjectId}[_{version}][___{subObjectType}_{subObjectId}],
+     *
+     *  where special characters in the individual tokens will get URL-encoded.
+     *
+     * @return URL-encoded version of this GUID
+     */
+    public String getURLEncoded() throws IOException {
+
+        String encodedGUID = toString("___", "_");
```

Comment:
If we're using "_" to separate the reference part, does that mean we need to make it an illegal character for object IDs?
Same question for subobject types.

Reply:
If I understand what you're saying, the ID transformation is one way?

Reply:
Yep

Reply:
Fair enough then.